### PR TITLE
KTXLoader.getSphericalHarmonics

### PR DESCRIPTION
**Pull request by alexey-pelykh**
Co-authored-by: Fabian Rueckert <fabi.rueckert@gmail.com>

### CODE REVIEW

**Comment by FabianRueckert** [comment]
@googlebot I consent.

File: android/filament-utils-android/src/main/java/com/google/android/filament/utils/KtxLoader.kt [hunk 1]
```diff
@@ -76,7 +76,14 @@ object KtxLoader {
         return Skybox(engine, nativeSkybox)
     }
 
+    fun getSphericalHarmonics(buffer: Buffer): FloatArray {
+        val floatBuffer = FloatBuffer.allocate(3 * 9)
```
Comment (romainguy) [inline android/filament-utils-android/src/main/java/com/google/android/filament/utils/KtxLoader.kt:80]: Why go through a `FloatBuffer`, why not use a regular array?

**Review by romainguy — COMMENTED** [review]
Could you explain why this change is needed?

**Comment by FabianRueckert** [comment]
> Could you explain why this change is needed?

To be able to separately load a KTX texture and spherical harmonics, that may then be modified on the client side, for example modulated using AR light estimation and used to create a filament indirect light from these modified values.
@romainguy

File: android/filament-utils-android/src/main/cpp/Utils.cpp [hunk 2]
```diff
@@ -73,6 +73,13 @@ static jlong nCreateSkybox(JNIEnv* env, jclass,
     return (jlong) Skybox::Builder().environment(cubemap).showSun(true).build(*engine);
 }
 
+static void nGetSphericalHarmonics(JNIEnv* env, jclass, jobject javaBuffer, jint remaining, jobject outBuffer) {
+    AutoBuffer buffer(env, javaBuffer, remaining);
+    KtxBundle bundle((const uint8_t*) buffer.getData(), buffer.getSize());
+    AutoBuffer outAutoBuffer(env, outBuffer, 3 * 9, true);
+    bundle.getSphericalHarmonics((float3*) outAutoBuffer.getData());
```
Comment (romainguy) [inline android/filament-utils-android/src/main/cpp/Utils.cpp:80]: What happens btw if the KTX doesn't contain spherical harmonics here?

**Review by romainguy — CHANGES_REQUESTED** [review]

**Comment by prideout** [comment]
LGTM if Romain's comments are addressed and it is rebased to the latest main (we named KtxLoader to KTXLoader)

**Review by prideout — APPROVED** [review]
LGTM if Romain's comments are addressed and it is rebased to the latest main (we named KtxLoader to KTXLoader)

File: android/filament-utils-android/src/main/java/com/google/android/filament/utils/KtxLoader.kt (hunk 1 above)
Comment (alexey-pelykh) [reply android/filament-utils-android/src/main/java/com/google/android/filament/utils/KtxLoader.kt]: Changed to `FloatArray`

File: android/filament-utils-android/src/main/cpp/Utils.cpp (hunk 2 above)
Comment (alexey-pelykh) [reply android/filament-utils-android/src/main/cpp/Utils.cpp]: Now it would return null. Btw, `nCreateIndirectLight` also does no checks.

**Pull request merged**